animations-tutorial.md

```diff
@@ -0,0 +1,589 @@
+to be called again. In `build`, the container changes size because its height
+and width now use `animation.value` instead of hardcoding 300.0.
+
+With these short changes, you’ve created your first animation in Flutter!
+
+## Simplifying with AnimatedComponent
+
+The code for `addListener` and `setState` is an obvious candidate for
+refactoring, which you’ll find in a helper class named
+[`AnimatedComponent`](http://docs.flutter.io/flutter/widgets/AnimatedComponent-class.html).
+The simplified code is shown below. LogoApp now derives from AnimatedComponent
+instead of from
+[`StatefulComponent`](http://docs.flutter.io/flutter/widgets/StatefulComponent-class.html), and
+your `AnimatedComponent` does not need to maintain a State object to hold the animation.
+
+You’ll notice that `LogoApp` passes the `Animation` object to the base class and
+uses `animation.value` to set the height and width of the container, so it works
+exactly the same as the previous example. This animation is just 26 lines!
+
+```dart
+import 'package:flutter/animation.dart';
+import 'package:flutter/material.dart';
+
+const kLogoUrl = "https://raw.githubusercontent.com/dart-lang/logos/master/logos_and_wordmarks/dart-logo.png";
+
+class LogoApp extends AnimatedComponent {
+  LogoApp({Key key, Animation<double> animation})
+      : super(key: key, animation: animation);
+
+  Widget build(BuildContext context) {
+    return new Center(
+        child: new Container(
+            margin: new EdgeDims.symmetric(vertical: 10.0),
+            height: animation.value,
+            width: animation.value,
+            child: new NetworkImage(src: kLogoUrl)));
+  }
+}
+
+void main() {
+  final AnimationController controller =
+      new AnimationController(duration: const Duration(milliseconds: 2000));
+  final Animation<double> animation =
+      new Tween(begin: 0.0, end: 300.0).animate(controller);
+  runApp(new LogoApp(animation: animation));
+  controller.forward();
+}
+```
+
+## Monitoring the progress of the animation
+
+It’s often helpful to know when an animation changes state, such as finishing,
+moving forward, or reversing. You can get notifications for this with
+`addStatusListener`.  Here’s a trivial change to the previous example to
+demonstrate these notifications: (changes noted in bold)
+
+<pre>
+void main() {
+  final AnimationController controller =
+      new AnimationController(duration: const Duration(milliseconds: 2000))
+       <b>..addStatusListener((state) => print("$state") )</b>;
+  final animation = new Tween(begin: 0.0, end: 300.0).animate(controller);
+  runApp(new LogoApp(animation));
+  controller.forward();
+}
+</pre>
+
+If you look at the logs after running this change, you’ll see lines like this as
+the animation progresses:
+
+```
+01-28 10:30:58.268 23497 23524 I flutter : AnimationStatus.forward
+...
+01-28 10:30:59.760 23497 23524 I flutter : AnimationStatus.completed
+```
```

Comment:
Good catch. The latest code doesn't print the timestamps, so let's match our output to the latest code:

```
I/flutter (21456): AnimationStatus.forward
I/flutter (21456): AnimationStatus.completed
```